fuels-abigen-macro/tests/harness.rs

```diff
@@ -83,9 +81,9 @@ async fn compile_bindings_from_inline_contract() {
         "#,
     );
 
-    let fuel_client = setup_local_node().await;
-
-    let contract_instance = SimpleContract::new(null_contract_id(), fuel_client);
+    let (provider, wallet) = setup_test_provider_and_wallet().await;
+    //`SimpleContract` is the name of the contract
+    let contract_instance = SimpleContract::new(null_contract_id(), provider.client, wallet);
```

Comment:
Just an idea: what do you think about changing `::new()` (from the `abigen`) to take in the provider instead of `FuelClient`?

Seems like we (and our users) would be passing `provider.client` a lot -- which feels clunky.

Reply:
I totally agree it seems much better!